library/src/main/java/com/jmolsmobile/landscapevideocapture/camera/NativeCamera.java

```diff
@@ -127,4 +130,34 @@ private boolean hasFrontFacingCamera() {
         }
         return false;
     }
+
+    public void setFlash(boolean isFlashOn){
+
+        if(camera != null) {
```

Comment:
When setFlash() is being called the camera should already be open (camera != null) and params should already be accessed before (params != null) hence it doesn't make any sense to check for this.

I prefer to program offensively: if something happens that you didn't expect your code should crash so you are aware of that. (otherwise it fails silently) In this case a consumer of setFlash() could call it and expect it to work, but internally the method would fail. Not checking for null, makes the contract more strict and enforces a single way to use this api.

Please remove null check for camera and params, the null check for flashmodes is valid

Reply:
Very interesting remark, I thought to do well, I take note and correct.